src/blas/backends/cublas/cublas_level3.cpp

```diff
@@ -387,7 +387,26 @@ inline cl::sycl::event gemm(Func func, cl::sycl::queue &queue, transpose transa,
                             int64_t m, int64_t n, int64_t k, T alpha, const T *a, int64_t lda,
                             const T *b, int64_t ldb, T beta, T *c, int64_t ldc,
                             const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
-    throw backend_unsupported_exception();
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+    overflow_check(m, n, k, lda, ldb, ldc);
+    auto done = queue.submit([&](cl::sycl::handler &cgh) {
+        int64_t num_events = dependencies.size();
+        for (int64_t i = 0; i < num_events; i++) {
+            cgh.depends_on(dependencies[i]);
+        }
+        cgh.interop_task([=](cl::sycl::interop_handler ih) {
+            auto sc     = CublasScopedContextHandler(queue);
+            auto handle = sc.get_handle(queue);
+            auto a_     = reinterpret_cast<const cuDataType *>(a);
+            auto b_     = reinterpret_cast<const cuDataType *>(b);
+            auto c_     = reinterpret_cast<cuDataType *>(c);
+            cublasStatus_t err;
+            CUBLAS_ERROR_FUNC(func, err, handle, get_cublas_operation(transa),
+                              get_cublas_operation(transb), m, n, k, (cuDataType *)&alpha, a_, lda,
+                              b_, ldb, (cuDataType *)&beta, c_, ldc);
+        });
+    });
+    return done;
```

Comment:
Would it be easier to read if we separate buffer and USM APIs in different files?

Reply:
Thanks for feedback. In most of the backends and header files (except mklgpu), Buffer and USM APIs are implemented in same file. Personally, I think, this is more concise and consistent with rest of code.

Reply:
I guess, I am more concerned about the consistency across the repo. If we agree on splitting the files, I would prefer doing it for all .cpp/.hpp files for all backends.

Reply:
yes the suggestion was a "generic" one to apply everywhere applicable